### ISSUE TITLE
Attaching disks on vSphere VM with govc

### ISSUE THREAD

**Issue by soekmego** [vmware/govmomi#1551]
I am having issues attaching an existing disk (vmdk) to a vSphere VM without an initial harddisk via govc. 

All necessary variables are exported already, like the datastore, credentials etc.

I uploaded a vmdk which was tested with VMWare Player (it boots up and I can log in) to the datastore on the ESXi datacenter without any errors. The command used for this task is:
```
govc datastore.upload
```  
Later on I created a VM without any disks and tried to attach the vmdk I uploaded previously via the following command:
```
govc vm.disk.attach -vm testvm -disk test_img/myimage.vmdk
```  
and this error-message occures once I try to power on the VM with the `govc vm.power -on=true testvm` command:
```
Powering on VirtualMachine:vm-389044... govc: File /vmfs/volumes/5ce7e498-26699e67-0545-3ca82a1204a0/testvm/testvm.vmdk was not found

```  
When checking the settings of the VM, the disk is shown as `testvm/testvm.vmdk` instead of `test_img/myimage.vmdk`.

When creating a VM via the GUI and attaching the previously uploaded vmdk manually, the VM boots up and works as intended. It is just the govc CLI tool which causes troubles.

Checking the datastore after creating the VM, the directory in which I uploaded the vmdk as well as the directory created for the VM are present.

When I remove/destroy the VM with the `govc vm.destroy testvm`, the uploaded disk gets deleted as well, even though it wasn't shown as attached in the VM settings.

**Comment by dougm** [comment]
You need to use `import.vmdk` rather than `datastore.upload`.  This takes care of the required format conversion process, the desktop vmdk format doesn't work on ESX.

How did you upload the vmdk using the GUI?  Maybe the GUI does the conversion process during that, but datastore.upload does not.

**Comment by soekmego** [comment]
I uploaded the vmdk via `govc datastore.upload`, created the VM using the GUI and chose the option to use an already existing disk. This gave me a working VM where I logged in to test it, powered it off and removed the VM with the disk from the datastore. I checked the node where I created the VM and the directory in the datastore to make sure everything is removed.

Then I uploaded the same vmdk from my local machine again via `govc datastore.upload`. This time I wanted to make everything via the console, to make sure everything works because I want to automate these steps later. 

I created a VM with `vm,create` and faced this error:
```
govc datastore.upload myimage.vmdk test_img/myimage.vmdk                           
[08-08-19 11:45:24] Uploading... OK

govc vm.create -m 2048 -c 2 -annotation="testing by gkn" -disk=test_img/myimage.vmdk testvm 

govc: File /vmfs/volumes/5ce7e498-26699e67-0545-3ca82a1204a0/testvm/testvm.vmdk was not found

```

I'll test the `import.vmdk` command tomorrow and report back.

**Comment by soekmego** [comment]
Like promised, here are the results of the `import.vmdk` command.

Running `govc import.vmdk myimage.vmdk test_img/myimage.vmdk` returned the following (very precise) error:
```
govc: vmdk: invalid format (must be streamOptimized)
The vmdk can be converted using one of:
  vmware-vdiskmanager -t 5 -r 'myimage.vmdk' new.vmdk
  qemu-img convert -O vmdk -o subformat=streamOptimized 'myimage.vmdk' new.vmdk
``` 
I converted the file as advised with the second command and tried to import the now stream-optimized vmdk.
`govc import.vmdk optimized.vmdk test_img/optimized.vmdk`
This is what I got:
```
[09-08-19 10:25:22] Uploading optimized.vmdk... OK
govc: File [datastore.node] test_img/optimized.vmdk/ was not found
```

It appears, that the datastore has troubles managing the files I upload. Weird thing is, I cant even delete the vmdk I uploaded yesterday anymore. It just says `Cant delete file optimized.vmdk`.

**Comment by dougm** [comment]
The 2nd arg is a directory, so try: `govc import.vmdk optimized.vmdk test_img`

As for the file you can't delete, is it still attached to a vm?

**Comment by soekmego** [comment]
Sorry for the late reply.

The disk I could not delete was not attached to a VM. However, I think this might have something to do with [the machine lock to prevent concurrent changes.](https://kb.vmware.com/s/article/10051) 
I removed the directory where said disk was placed by me, which deleted the disk for good.

As for the import statement, a directory without a filename now works.
The [USAGE.md](https://github.com/vmware/govmomi/blob/master/govc/USAGE.md) already states that the 2nd arg as TARGET but for some reason I figured I should pass a name for the file, like the `govc datastore.upload` command. 

I'll gladly contribute to some of the examples in the USAGE.md file to prevent future confusion by other beginners, once I get back home.

Thank you for taking your time to help me.
